cirq/devices/noise_model.py

```diff
@@ -119,6 +76,25 @@ def noisy_operation(self, operation: 'cirq.Operation') -> 'cirq.OP_TREE':
             An OP_TREE corresponding to the noisy operations implementing the
             noisy version of the given operation.
         """
+        return operation
+
+    def apply_noise(self, input_circuit: 'cirq.Circuit') -> 'cirq.Circuit':
```

Comment:
Rename this to `noisy_circuit_moments` and have it return `self.noisy_moments(circuit, sorted(circuit.all_qubits()))`.

We should avoid having a `noisy_circuit` method that returns Circuit, because it is incapable of maintaining the moment structure.

Reply:
The goal here was to address #1610 - how to get `noisy_*` methods into a circuit.

The original default implementations of `noisy_moments`, `noisy_moment`, and `noisy_operation` have been wrapped up into this loop